### PR TITLE
Task #166301 chore: Merge API Plugin in release-1.4.5 branch

### PR DESCRIPTION
**Pull request by deepgawade89**

### CODE REVIEW

File: plugins/api/tjfields/tjfields.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+defined('_JEXEC') or die( 'Restricted access');
+
+use Joomla\CMS\Factory;
+/**
+ * Base Class for api plugin
+ *
+ * @package     Tjfields
+ * @subpackage  component
```
Comment (ankush-maherwal) [inline plugins/api/tjfields/tjfields.php:18]: @deepa-g subpackage is API plugin

File: plugins/api/tjfields/tjfields/cities.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+class TjfieldsApiResourceCities extends ApiResource
+{
+	/**
+	 *  API Plugin for get method
+	 *
+	 * @return  void
+	 *
+	 * @since   _DEPLOY_VERSION_
+	 */
+	public function post()
```
Comment (ankush-maherwal) [inline plugins/api/tjfields/tjfields/cities.php:33]: @deepa-g this should be get() method as its returning the data

File: plugins/api/tjfields/tjfields/regions.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	 *  API Plugin for get method
+	 *
+	 * @return  void
+	 *
+	 * @since   _DEPLOY_VERSION_
+	 */
+	public function post()
+	{
+		$input           = Factory::getApplication()->input;
+		$id              = $input->getInt('country_id', 0);
```
Comment (ankush-maherwal) [inline plugins/api/tjfields/tjfields/regions.php:36]: @deepa-g Are we using all these APIs to get the list only? It seems that we cant get data of a particular country, city, or region.

File: script.tjfields.php [hunk 4]
```diff
@@ -30,6 +30,14 @@ class Com_TjfieldsInstallerScript
 			'site' => array(
 					'mod_tjfields_search' => array('tj-filters-mod-pos', 1)
 						)
+		),
+		'plugins'=>array(
+			'privacy'=>array(
+				'tjfields'=>0,
```
Comment (ankush-maherwal) [inline script.tjfields.php:36]: @deepa-g please check the indentation of the code

File: script.tjfields.php [hunk 5]
```diff
@@ -255,6 +263,63 @@ private function _installSubextensions($parent)
 			}
 		}
 
+		// Plugins installation
+		if (count($this->installation_queue['plugins']))
```
Comment (ankush-maherwal) [inline script.tjfields.php:267]: @deepa-g We can avoid this if we use the plugins as the sub-extension of the main package

File: plugins/api/tjfields/tjfields/cities.php (hunk 2 above)
Comment (deepgawade89) [reply plugins/api/tjfields/tjfields/cities.php:33]: As per discussion with Tushar, he was written this post() method specifically. @manojLondhe Can you please check this?

**Pull request closed without merging**